src/transforms/sample/transform.rs

```diff
@@ -12,15 +12,44 @@ use crate::{
 use vector_lib::lookup::lookup_v2::OptionalValuePath;
 use vector_lib::lookup::OwnedTargetPath;
 
+/// Exists only for backwards compatability purposes so that the value of sample_rate_key is
+/// consistent after the internal implementation of the Sample class was modified to work in terms
+/// of percentages
+#[derive(Clone, Copy, Debug)]
+pub enum SampleRate {
+    OneOverN(u64),
```

Comment:
Any reason not to call this just `Rate` to match the config setting? Maybe call the enum something like `Mode` since it changes the mode of operation of the sampler.